docs-developer/build.gradle.kts

```diff
@@ -1,16 +1,98 @@
 /*
  * Copyright 2014-2024 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
  */
+import com.pswidersk.gradle.python.VenvTask
+import org.gradle.api.tasks.PathSensitivity.RELATIVE
+import org.jetbrains.kotlin.util.parseSpaceSeparatedArgs
 
 plugins {
-    id("ru.vyarus.mkdocs") version "2.4.0"
+    id("com.pswidersk.python-plugin") version "2.4.0"
 }
 
-if (project.version.toString().endsWith("-SNAPSHOT")) {
-    // Do not generate the root index.html file with the redirect
```

Comment:
Unfortunately, I can't test it myself now and don't wanna delay the review, so I assume you've understood what this hack does and the new plugin does everything in the same way as the old one: 
* For non-snapahot versions, it should build and commit a directory with the version name and create a root index.html file that redirects to it, so that `kotlin.github.io/dokka` always redirects to the latest docs
* For snapshot versions, it should just build and commit a directory with the version name, no redirects (useful for when users use dev/rc/preview versions and for us to review upcoming changes)

Reply:
Ahh okay, I missed that. I saw that the redirecting index.html was a ru.vyarus.mkdocs specific issue, and I assumed that Dokka config was disabling this feature.

It sounds like a pretty simple algorithm to set up in Gradle logic, so I'll set it up. It's a shame that ru.vyarus.mkdocs is incompatible with Config Cache though, because less custom code is better!

Reply:
Implemented, and tested by manually changing the version in `gradle.properties`, running `./gradlew buildMkDocsSite`, and checking in `docs-developer/build/mkdocs/`

#### 2.0.0 

<img width="201" alt="image" src="https://github.com/Kotlin/dokka/assets/152864218/496953c9-6b64-464f-8576-33694730185b">


#### 2.0.0-SNAPSHOT

<img width="204" alt="image" src="https://github.com/Kotlin/dokka/assets/152864218/ef67b696-ebb3-4397-a4e3-cd7d1388c776">